test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -4819,10 +4819,43 @@ class Comm(PBSService):
     PBS ``Comm`` configuration and control
     """
 
-    logger = logging.getLogger(__name__)
-    logprefix = 'pbs_comm: '
+    dflt_attributes = {}
 
-    conf_to_cmd_map = {'PBS_COMM_ROUTERS': '-r', 'PBS_COMM_THREADS': '-t'}
+    def __init__(self, name=None, attrs={}, pbsconf_file=None, diagmap={},
+                 diag=None, server=None, db_access=None):
+        """
+        name - The hostname of the Comm. Defaults to current hostname.
+        attrs - Dictionary of attributes to set, these will override defaults.
+        pbsconf_file - path to config file to parse for PBS_HOME, PBS_EXEC, etc
+        diagmap - A dictionary of PBS objects (node,server,etc) to mapped files
+        from PBS diag directory
+        diag - path to PBS diag directory (This will overrides diagmap)
+        server - A PBS server instance to which this Comm is associated
+        db_acccess- set to either file containing credentials to DB access or
+        dictionary containing {'dbname':...,'user':...,'port':...}
+        """
+        self.logger = logging.getLogger(__name__)
+        if server is not None:
+            self.server = server
+            if diag is None and self.server.diag is not None:
+                diag = self.server.diag
+            if (len(diagmap) == 0) and (len(self.server.diagmap) != 0):
+                diagmap = self.server.diagmap
+        else:
+            self.server = Server(name, pbsconf_file=pbsconf_file,
+                                 db_access=db_access, diag=diag,
+                                 diagmap=diagmap)
+        PBSService.__init__(self, name, attrs, self.dflt_attributes,
```

Comment:
This part of the code is written the way as for other PBS daemons. Since this suggested change has to be done for all PBS daemons as well and is out of scope of this PR,  I have logged below JIRA ticket:
PP-714: In PTL PBS daemon class definition init methods need improvements

This ticket also includes the update to use super() for referring to base class.